// @uneed/jsbridge-runtime/modules/album.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Album = void 0;
var native_1 = require("../native");
var AlbumModule = /** @class */ (function (_super) {
    __extends(AlbumModule, _super);
    function AlbumModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AlbumModule.prototype.addPhoto = function (json) {
        return this.native.callNativeFlat('Album', 'addPhoto', json);
    };
    AlbumModule.prototype.addShareMem = function (albumid, isCreator, key) {
        return this.native.callNativeFlat('Album', 'addShareMem', { albumid: albumid, isCreator: isCreator, key: key });
    };
    AlbumModule.prototype.saveFromAlbum = function (json) {
        return this.native.callNativeFlat('Album', 'saveFromAlbum', json);
    };
    AlbumModule.prototype.shareToSystem = function (json) {
        return this.native.callNativeFlat('Album', 'shareToSystem', json);
    };
    AlbumModule.prototype.playAudio = function (json) {
        return this.native.callNativeFlat('Album', 'playAudio', json);
    };
    AlbumModule.prototype.openProfile = function (params) {
        return this.native.callNativeFlat('Album', 'openProfile', params);
    };
    AlbumModule.prototype.refreshAlbum = function (albumid) {
        return this.native.callNativeFlat('Album', 'refreshAlbum', { albumid: albumid });
    };
    AlbumModule.prototype.preview = function (params) {
        return this.native.callNativeFlat('Album', 'preview', params);
    };
    AlbumModule.prototype.recording = function (params) {
        return this.native.callNativeFlat('Album', 'recording', params);
    };
    AlbumModule.prototype.playRecord = function (params) {
        return this.native.callNativeFlat('Album', 'playRecord', params);
    };
    AlbumModule.prototype.geTIsSetPassword = function () {
        return this.native.callNativeFlat('Album', 'geTIsSetPassword', {});
    };
    AlbumModule.prototype.goToSetPassword = function () {
        return this.native.callNativeFlat('Album', 'goToSetPassword', {});
    };
    AlbumModule.prototype.shareWithOthers = function (json) {
        return this.native.callNativeFlat('Album', 'shareWithOthers', json);
    };
    AlbumModule.prototype.cloudIsFull = function () {
        return this.native.callNativeFlat('Album', 'cloudIsFull');
    };
    return AlbumModule;
}(native_1.ModuleBase));
exports.Album = new AlbumModule();
native_1.addJSModule('Album', exports.Album);
