// @uneed/jsbridge-runtime/modules/api.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.API = void 0;
var native_1 = require("../native");
var ApiModule = /** @class */ (function (_super) {
    __extends(ApiModule, _super);
    function ApiModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ApiModule.prototype.request = function (params) {
        if (!params.type || !/POST/i.test(params.type)) {
            params.type = 'GET';
        }
        params.type = params.type.toUpperCase();
        return this.native.callNative({
            module: 'API',
            method: 'request',
            args: params,
        });
    };
    ApiModule.prototype.uploadFile = function (params) {
        return this.native.callNative({
            module: 'API',
            method: 'uploadFile',
            args: params,
        });
    };
    return ApiModule;
}(native_1.ModuleBase));
exports.API = new ApiModule();
native_1.addJSModule('API', exports.API);
