// @uneed/jsbridge-runtime/modules/video.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Video = void 0;
var native_1 = require("../native");
var VideoModule = /** @class */ (function (_super) {
    __extends(VideoModule, _super);
    function VideoModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    VideoModule.prototype.play = function (params) {
        this.native.callNativeFlat('Video', 'play', params);
    };
    VideoModule.prototype.pause = function () {
        this.native.callNativeFlat('Video', 'pause');
    };
    VideoModule.prototype.resume = function () {
        this.native.callNativeFlat('Video', 'resume');
    };
    VideoModule.prototype.stop = function () {
        this.native.callNativeFlat('Video', 'stop');
    };
    return VideoModule;
}(native_1.ModuleBase));
exports.Video = new VideoModule();
native_1.addJSModule('Video', exports.Video);
