// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/app.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.App = void 0;
var native_1 = require("../native");
function uagte(version, ua) {
    var compare = function (a, b) {
        var result;
        var pre = a.split('.');
        var after = b.split('.');
        var len = pre.length > after.length ? pre.length : after.length;
        for (var i = 0; i < len; i++) {
            var preInt = parseInt(pre[i]) || 0;
            var afterInt = parseInt(after[i]) || 0;
            if (preInt > afterInt) {
                result = 1;
                break;
            }
            else if (preInt < afterInt) {
                result = -1;
                break;
            }
            else {
                if (i == len - 1) {
                    result = 0;
                }
            }
        }
        return result;
    };
    var uaVersion = ua.split(' ')[0].replace('UneedGroup/', '');
    var result = compare(uaVersion, version);
    return result >= 0;
}
var AppModule = /** @class */ (function (_super) {
    __extends(AppModule, _super);
    function AppModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AppModule.prototype.stopCatch = function (params) {
        return this.native.callNativeFlat('App', 'stopCatch', params);
    };
    AppModule.prototype.getAppUA = function () {
        return this.native.callNativeFlat('App', 'getAppUA');
    };
    AppModule.prototype.getAppUASync = function () {
        return this.native.callNativeSync('App', 'getAppUASync');
    };
    // greate than or equal
    AppModule.prototype.versionGTE = function (version) {
        return __awaiter(this, void 0, void 0, function () {
            var ua, yes;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.getAppUA()];
                    case 1:
                        ua = _a.sent();
                        yes = uagte(version, ua);
                        if (yes) {
                            return [2 /*return*/, Promise.resolve(true)];
                        }
                        else {
                            return [2 /*return*/, Promise.resolve(false)];
                        }
                        return [2 /*return*/];
                }
            });
        });
    };
    AppModule.prototype.getClipBoardContent = function () {
        return this.native.callNativeFlat('App', 'getClipBoardContent');
    };
    AppModule.prototype.getProfile = function (uids, gids, type) {
        return this.native.callNativeFlat('App', 'getProfile', { gids: gids, uids: uids, type: type || 'profile' });
    };
    AppModule.prototype.setClipBoardContent = function (data) {
        return this.native.callNativeFlat('App', 'setClipBoardContent', data);
    };
    AppModule.prototype.getDeviceInfo = function () {
        return this.native.callNativeFlat('App', 'getDeviceInfo');
    };
    // 返回{ state: string }   wifi none cellular unknown等状态
    AppModule.prototype.getNetwork = function () {
        return this.native.callNativeFlat('App', 'getNetwork');
    };
    AppModule.prototype.getNetworkSync = function () {
        return this.native.callNativeSync('App', 'getNetworkSync');
    };
    AppModule.prototype.scan = function () {
        return this.native.callNativeFlat('App', 'scan');
    };
    AppModule.prototype.getUserInfo = function () {
        return this.native.callNativeFlat('App', 'getUserInfo');
    };
    AppModule.prototype.updateUserInfo = function () {
        this.native.callNativeFlat('App', 'updateUserInfo');
    };
    AppModule.prototype.getDisplayName = function (params) {
        return this.native.callNativeFlat('App', 'getDisplayName', params);
    };
    AppModule.prototype.getStepNum = function () {
        return this.native.callNativeFlat('App', 'getStepNum');
    };
    AppModule.prototype.getLocation = function () {
        return this.native.callNativeFlat('App', 'getLocation');
    };
    AppModule.prototype.shareLocation = function (params) {
        return this.native.callNativeFlat('App', 'shareLocation', params);
    };
    // 隐私设置里的 允许后台上传位置
    AppModule.prototype.sharePrivateLocation = function (params) {
        return this.native.callNativeFlat('App', 'sharePrivateLocation', params);
    };
    AppModule.prototype.openScheme = function (url) {
        return this.native.callNativeFlat('App', 'openScheme', { url: url });
    };
    AppModule.prototype.openSelectPoi = function (geo) {
        return this.native.callNativeFlat('App', 'openSelectPoi', {
            geo: geo,
        });
    };
    //调起投屏 向客户端传递视频信息
    AppModule.prototype.openSelectTV = function (params) {
        return this.native.callNativeFlat('App', 'openSelectTV', params);
    };
    // 阅读器页面打开文件处理页面
    AppModule.prototype.openFilePage = function (params) {
        return this.native.callNativeFlat('App', 'openFilePage', params);
    };
    AppModule.prototype.getPackageManifest = function (id) {
        return this.native.callNativeFlat('App', 'getPackageManifest', {
            id: id,
        });
    };
    AppModule.prototype.selectCity = function () {
        return this.native.callNativeFlat('App', 'selectCity');
    };
    AppModule.prototype.selectHometown = function () {
        return this.native.callNativeFlat('App', 'selectHometown');
    };
    AppModule.prototype.openSelectAlbum = function (params) {
        return this.native.callNativeFlat('App', 'openSelectAlbum', params);
    };
    AppModule.prototype.openSelectCloud = function (params) {
        return this.native.callNativeFlat('App', 'openSelectCloud', params);
    };
    AppModule.prototype.createGroup = function () {
        return this.native.callNativeFlat('App', 'createGroup');
    };
    AppModule.prototype.getUserProfile = function (uid, type) {
        return this.native.callNativeFlat('App', 'getUserProfileByUid', {
            uid: uid,
            type: type || 'profile',
        });
    };
    AppModule.prototype.refreshUserProfile = function (uids, type) {
        return this.native.callNativeFlat('App', 'refreshUserProfile', {
            uids: uids,
            type: type,
        });
    };
    AppModule.prototype.getGroupProfile = function (gid) {
        return this.native.callNativeFlat('App', 'getGroupProfileByGid', {
            gid: gid,
        });
    };
    AppModule.prototype.shareImage = function (filePath) {
        return this.native.callNativeFlat('App', 'shareImage', { filePath: filePath });
    };
    AppModule.prototype.share = function (params) {
        return this.native.callNativeFlat('App', 'share', params);
    };
    AppModule.prototype.shareCard = function (params) {
        return this.native.callNativeFlat('App', 'shareCard', params);
    };
    AppModule.prototype.systemShare = function (params) {
        return this.native.callNativeFlat('App', 'systemShare', params);
    };
    AppModule.prototype.openForward = function (params) {
        if (params === void 0) { params = {}; }
        return this.native.callNativeFlat('App', 'openForward', params);
    };
    AppModule.prototype.openSelectCvs = function (params) {
        if (params === void 0) { params = {}; }
        return this.native.callNativeFlat('App', 'openSelectCvs', params);
    };
    AppModule.prototype.selectGroupMember = function (params) {
        if (params === void 0) { params = {}; }
        return this.native.callNativeFlat('App', 'selectGroupMember', params);
    };
    AppModule.prototype.showAddFriendPanel = function (params) {
        return this.native.callNativeFlat('App', 'showAddFriendPanel', params);
    };
    AppModule.prototype.checkUpdate = function () {
        return this.native.callNativeFlat('App', 'checkUpdate');
    };
    // 'weibo ins linkin...'
    AppModule.prototype.bindThirdPart = function (data) {
        return this.native.callNativeFlat('App', 'bindThirdPart', data);
    };
    AppModule.prototype.updateNativeInfo = function (data) {
        if (data === void 0) { data = {}; }
        return this.native.callNativeFlat('App', 'updateNativeInfo', data);
    };
    // 获取android机型 返回机型对应name和version
    AppModule.prototype.getRom = function () {
        return this.native.callNativeFlat('App', 'getRom');
    };
    // 图片创建pdf
    AppModule.prototype.createPdf = function (urls) {
        return this.native.callNativeFlat('App', 'createPdf', urls);
    };
    // 获取手机环境参数
    AppModule.prototype.appEnv = function () {
        return this.native.callNativeFlat('App', 'appEnv');
    };
    // 获取手机环境参数
    AppModule.prototype.getThemeModel = function () {
        return this.native.callNativeFlat('App', 'getThemeModel');
    };
    // 调用客户端上传下载文件至本地 云盘 相册
    AppModule.prototype.download = function (params) {
        return this.native.callNative({
            module: 'App',
            method: 'download',
            args: params,
        });
    };
    // 打开客户端的传输队列页面 type: 0 是上传 1 是下载
    AppModule.prototype.toTransferQueue = function (params) {
        return this.native.callNativeFlat('App', 'toTransferQueue', params);
    };
    //获取用户名 优先备注 然后昵称
    AppModule.prototype.getUserName = function (uid) {
        return this.native.callNativeFlat('App', 'getUserName', { uid: uid });
    };
    AppModule.prototype.shareWithOthers = function (json) {
        return this.native.callNativeFlat('App', 'shareWithOthers', json);
    };
    return AppModule;
}(native_1.ModuleBase));
exports.App = new AppModule();
native_1.addJSModule('App', exports.App);
