// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/audio.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audio = void 0;
var native_1 = require("../native");
var AudioModule = /** @class */ (function (_super) {
    __extends(AudioModule, _super);
    function AudioModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AudioModule.prototype.download = function (url) {
        return this.native.callNativeFlat('Audio', 'download', url);
    };
    AudioModule.prototype.play = function (params) {
        this.native.callNativeFlat('Audio', 'play', params);
    };
    AudioModule.prototype.resume = function (params) {
        this.native.callNativeFlat('Audio', 'resume', params);
    };
    AudioModule.prototype.pause = function (channel) {
        this.native.callNativeFlat('Audio', 'pause', channel);
    };
    AudioModule.prototype.stop = function (channel) {
        this.native.callNativeFlat('Audio', 'stop', channel);
    };
    AudioModule.prototype.getState = function (channel) {
        return this.native.callNativeFlat('Audio', 'getState', channel);
    };
    return AudioModule;
}(native_1.ModuleBase));
exports.Audio = new AudioModule();
native_1.addJSModule('Audio', exports.Audio);
