// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/chat.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Chat = void 0;
var native_1 = require("../native");
var ChatModule = /** @class */ (function (_super) {
    __extends(ChatModule, _super);
    function ChatModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    //添加消息提醒
    ChatModule.prototype.sendText = function (params) {
        return this.native.callNativeFlat('Chat', 'sendText', params);
    };
    ChatModule.prototype.sendImage = function (filePath) {
        return this.native.callNativeFlat('Chat', 'sendImage', { filePath: filePath });
    };
    // 发送表情 结构暂定
    ChatModule.prototype.sendSticker = function (params) {
        return this.native.callNativeFlat('Chat', 'sendSticker', params);
    };
    // url 是与你头像结构
    ChatModule.prototype.setChatAvatar = function (params) {
        return this.native.callNativeFlat('Chat', 'setChatAvatar', params);
    };
    ChatModule.prototype.setBackground = function (params) {
        return this.native.callNativeFlat('Chat', 'setBackground', params);
    };
    ChatModule.prototype.uploadRecentMessages = function (params) {
        return this.native.callNativeFlat('Chat', 'uploadRecentMessages', params);
    };
    return ChatModule;
}(native_1.ModuleBase));
exports.Chat = new ChatModule();
native_1.addJSModule('Chat', exports.Chat);
