// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/image.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Image = void 0;
var native_1 = require("../native");
var ImageModule = /** @class */ (function (_super) {
    __extends(ImageModule, _super);
    function ImageModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ImageModule.prototype.saveBase64ToFile = function (base64) {
        return this.native.callNativeFlat('Image', 'saveBase64ToFile', { base64: base64 });
    };
    ImageModule.prototype.saveToSystemAlbum = function (filePath, url) {
        return this.native.callNativeFlat('Image', 'saveToSystemAlbum', { filePath: filePath, url: url });
    };
    ImageModule.prototype.openSystemCamera = function (params) {
        return this.native.callNative({
            module: 'Image',
            method: 'openSystemCamera',
            args: params,
        });
    };
    ImageModule.prototype.pickSystemImage = function (params) {
        if (!params.max) {
            params.max = 0;
        }
        return this.native.callNative({
            module: 'Image',
            method: 'pickSystemImage',
            args: params,
        });
    };
    ImageModule.prototype.previewImage = function (params) {
        return this.native.callNative({
            module: 'Image',
            method: 'previewImage',
            args: params,
        });
    };
    ImageModule.prototype.previewWebImages = function (params) {
        this.native.callNativeFlat('Image', 'previewWebImages', params);
    };
    ImageModule.prototype.previewAlbumImage = function (params) {
        return this.native.callNative({
            module: 'Image',
            method: 'previewAlbumImage',
            args: params,
        });
    };
    return ImageModule;
}(native_1.ModuleBase));
exports.Image = new ImageModule();
native_1.addJSModule('Image', exports.Image);
