// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/nimo.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Nimo = void 0;
var native_1 = require("../native");
var NimoModule = /** @class */ (function (_super) {
    __extends(NimoModule, _super);
    function NimoModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    // Nimo透传过来的信息
    NimoModule.prototype.onData = function (cb) {
        this.native.callNative({
            module: 'Nimo',
            method: 'onData',
            args: cb,
        });
    };
    return NimoModule;
}(native_1.ModuleBase));
exports.Nimo = new NimoModule();
native_1.addJSModule('Nimo', exports.Nimo);
