// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/permission.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Permission = exports.PermissionStatus = exports.PermissionType = void 0;
var native_1 = require("../native");
var PermissionType;
(function (PermissionType) {
    PermissionType[PermissionType["stepCount"] = 1] = "stepCount";
    PermissionType[PermissionType["locationWhenInUse"] = 2] = "locationWhenInUse";
    PermissionType[PermissionType["locationAlways"] = 3] = "locationAlways";
})(PermissionType = exports.PermissionType || (exports.PermissionType = {}));
/*!
 定义权限状态，目的是得到确定的Int用于跨语言传送
 */
var PermissionStatus;
(function (PermissionStatus) {
    PermissionStatus[PermissionStatus["unavailable"] = 1] = "unavailable";
    PermissionStatus[PermissionStatus["denied"] = 2] = "denied";
    PermissionStatus[PermissionStatus["authorized"] = 3] = "authorized";
    PermissionStatus[PermissionStatus["notDetermined"] = 4] = "notDetermined";
    PermissionStatus[PermissionStatus["unknow"] = 5] = "unknow";
    PermissionStatus[PermissionStatus["error"] = 6] = "error";
    PermissionStatus[PermissionStatus["authorizedAll"] = 7] = "authorizedAll";
    PermissionStatus[PermissionStatus["disabled"] = 8] = "disabled";
})(PermissionStatus = exports.PermissionStatus || (exports.PermissionStatus = {}));
var PermissionModule = /** @class */ (function (_super) {
    __extends(PermissionModule, _super);
    function PermissionModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PermissionModule.prototype.request = function (type) {
        return this.native.callNative({
            module: 'Permission',
            method: 'request',
            args: {
                type: type,
            },
        });
    };
    PermissionModule.prototype.requestWithTips = function (type) {
        return this.native.callNative({
            module: 'Permission',
            method: 'requestWithTips',
            args: {
                type: type,
            },
        });
    };
    return PermissionModule;
}(native_1.ModuleBase));
exports.Permission = new PermissionModule();
native_1.addJSModule('Permission', exports.Permission);
