// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/push.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Push = exports.PushShowGuideStatus = exports.PushType = void 0;
var native_1 = require("../native");
// 设置类型
var checkStatus;
(function (checkStatus) {
    checkStatus[checkStatus["Unknow"] = 0] = "Unknow";
    checkStatus[checkStatus["On"] = 1] = "On";
    checkStatus[checkStatus["Off"] = 2] = "Off";
})(checkStatus || (checkStatus = {}));
var GuideType;
(function (GuideType) {
    GuideType[GuideType["None"] = 0] = "None";
    GuideType[GuideType["Switch"] = 1] = "Switch";
    GuideType[GuideType["Navigation"] = 2] = "Navigation";
    GuideType[GuideType["Default"] = 3] = "Default";
})(GuideType || (GuideType = {}));
var PushType;
(function (PushType) {
    PushType[PushType["Notice"] = 1] = "Notice";
    PushType[PushType["AutoStart"] = 2] = "AutoStart";
    PushType[PushType["HighEnergy"] = 3] = "HighEnergy";
    PushType[PushType["Frozen"] = 4] = "Frozen";
    PushType[PushType["TaskLock"] = 5] = "TaskLock";
    PushType[PushType["HomeNotify"] = 6] = "HomeNotify";
    PushType[PushType["YUniNotice"] = 7] = "YUniNotice";
    PushType[PushType["SystemService"] = 8] = "SystemService";
})(PushType = exports.PushType || (exports.PushType = {}));
// 跳转到引导页
var PushShowGuideStatus;
(function (PushShowGuideStatus) {
    PushShowGuideStatus[PushShowGuideStatus["System"] = 1] = "System";
    PushShowGuideStatus[PushShowGuideStatus["Switch"] = 2] = "Switch";
})(PushShowGuideStatus = exports.PushShowGuideStatus || (exports.PushShowGuideStatus = {}));
var PushModule = /** @class */ (function (_super) {
    __extends(PushModule, _super);
    function PushModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    // 返回 GuideType
    PushModule.prototype.getGuide = function (type) {
        return this.native.callNativeFlat('Push', 'getGuide', { type: type });
    };
    // 返回 checkStatus
    PushModule.prototype.check = function (type) {
        return this.native.callNativeFlat('Push', 'check', { type: type });
    };
    PushModule.prototype.setStatus = function (params) {
        return this.native.callNativeFlat('Push', 'setStatus', params);
    };
    PushModule.prototype.showGuide = function (type) {
        return this.native.callNativeFlat('Push', 'showGuide', { type: type });
    };
    return PushModule;
}(native_1.ModuleBase));
exports.Push = new PushModule();
native_1.addJSModule('Push', exports.Push);
