// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/redpacket.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Redpacket = void 0;
var native_1 = require("../native");
var RedpacketModule = /** @class */ (function (_super) {
    __extends(RedpacketModule, _super);
    function RedpacketModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RedpacketModule.prototype.send = function (params) {
        return this.native.callNative({
            module: 'Redpacket',
            method: 'send',
            args: params,
        });
    };
    return RedpacketModule;
}(native_1.ModuleBase));
exports.Redpacket = new RedpacketModule();
native_1.addJSModule('Redpacket', exports.Redpacket);
