// _@uneed_jsbridge-webview@4.6.7@@uneed/jsbridge-webview/modules/userDefaults.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserDefaults = void 0;
var native_1 = require("../native");
var UserDefaulsModule = /** @class */ (function (_super) {
    __extends(UserDefaulsModule, _super);
    function UserDefaulsModule() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UserDefaulsModule.prototype.bool = function (key, defaultVal) {
        return this.native.callNativeFlat('UserDefaults', 'getBool', {
            key: key,
            defaultVal: defaultVal,
        });
    };
    UserDefaulsModule.prototype.int = function (key, defaultVal) {
        return this.native.callNativeFlat('UserDefaults', 'getInt', {
            key: key,
            defaultVal: defaultVal,
        });
    };
    UserDefaulsModule.prototype.string = function (key, defaultVal) {
        return this.native.callNativeFlat('UserDefaults', 'getString', {
            key: key,
            defaultVal: defaultVal,
        });
    };
    UserDefaulsModule.prototype.setBool = function (key, value) {
        return this.native.callNativeFlat('UserDefaults', 'setBool', {
            key: key,
            value: value,
        });
    };
    UserDefaulsModule.prototype.setInt = function (key, value) {
        return this.native.callNativeFlat('UserDefaults', 'setInt', {
            key: key,
            value: value,
        });
    };
    UserDefaulsModule.prototype.setString = function (key, value) {
        return this.native.callNativeFlat('UserDefaults', 'setString', {
            key: key,
            value: value,
        });
    };
    return UserDefaulsModule;
}(native_1.ModuleBase));
exports.UserDefaults = new UserDefaulsModule();
native_1.addJSModule('UserDefaults', exports.UserDefaults);
